test/Main.hs

```diff
@@ -113,3 +116,8 @@ roundTrip name encode decode =
             case Decode.parse decode (BL.toStrict bytes) of
                 Left _ -> error "Could not decode encoded message"
                 Right x' -> x === x'
+
+decodeNonsense :: TestTree
+decodeNonsense = HU.testCase "Decoding a nonsensical string fails." $ do
+  let decoded = Decode.parse (one Decode.fixed64 0 `at` fieldNumber 1) "test"
+  decoded HU.@?= Left (Decode.BinaryError "Failed reading: Encountered bytes that aren't valid key/value pairs.\nEmpty call stack\n")
```

Comment:
This test looks possibly fragile, in that it might break for different, even minor, versions of `cereal`. Should we use `isLeft` or some pattern instead?